modules/imgproc/src/hal_replacement.hpp

```diff
@@ -763,6 +763,26 @@ inline int hal_ni_medianBlur(const uchar* src_data, size_t src_step, uchar* dst_
 #define cv_hal_medianBlur hal_ni_medianBlur
 //! @endcond
 
+/**
+   @brief Calculate bilateral filter. See https://homepages.inf.ed.ac.uk/rbf/CVonline/LOCAL_COPIES/MANDUCHI1/Bilateral_Filtering.html
+   @param src_data Source image data
+   @param src_step Source image step
+   @param dst_data Destination image data
+   @param dst_step Destination image step
+   @param width Source image width
+   @param height Source image height
+   @param depth Depths of source and destination image
+   @param cn Number of channels
+   @param d Diameter of each pixel neighborhood that is used during filtering. If it is non-positive, it is computed from sigmaSpace.
+*/
+inline int hal_ni_bilateralFilter(const uchar* src_data, size_t src_step, uchar* dst_data, size_t dst_step,
```

Comment:
the API is a bit confusing; semantically, it should take pointers of any type (U8 and F32 at least, maybe F16/BF16 in the future), but it uses `uchar*`. Suggest to use `void*`.

Reply:
I reviewed existing HAL interfaces and it's done in the same way everywhere. Examples: hal_ni_integral, hal_ni_resize. I added information about supported types to description.